app/controllers/DepartureMessagesController.scala

```diff
@@ -86,7 +98,7 @@ class DepartureMessagesController @Inject() (
                                 request.body
                               )
                             )
-                          ).withHeaders(LOCATION -> routes.DepartureMessagesController.getDepartureMessage(departureId, messageId).urlWithContext)
+                          ).withHeaders(LOCATION -> routing.routes.DeparturesRouter.getMessage(departureId.toString, messageId.toString).urlWithContext)
```

Comment:
I think we should be using .value for case classes that extend anyval rather than toString, one just retrieves the value already in memory and the other executes a function to derive a value each time it's called?

Reply:
Truth be told, I don't know why I used toString... my bad.

Reply:
Oh, yes, I do, now I looked at it. In P4, these are interger backed values, not string. So we need to do the conversion anyway.

Reply:
But wouldn't it already be a string when you are retrieving the value from the case class?

Reply:
I get the initial conversion to set the departureId value

Reply:
No - it's an integer backed class.

Reply:
https://github.com/hmrc/common-transit-convention-traders/blob/acbc65b663825309f72783a4c27d0028cab56fd7/app/models/domain/DepartureId.scala#L21

Reply:
(didn't mean to click the resolve button!)

Reply:
We have two DepartureIds? Sorry if I'm being slow: https://github.com/hmrc/common-transit-convention-traders/pull/282/files#diff-f6c5dac1148800169cd955c75ed3164515e472a20629e842742b25aefb56f8a7

Reply:
Also, shouldn't we make the def toString in the case class a val, so we only have to calculate the string value once?

Reply:
We do - the P4 is an integer backed one, the P5 one is a string backed one.

We can't make the def a val, because then it can't be an AnyVal.

Reply:
Gotcha, question time is over ;) you can mark this as resolved now :D